stable/datadog/templates/agent-service-metrics.yaml

```
@@ -9,6 +9,7 @@ metadata:
     release: {{ .Release.Name | quote }}
     heritage: {{ .Release.Service | quote }}
 spec:
+  type: {{ .Values.clusterAgent.service.type }}
```

Default value of type is `clusterIP`, so I think we can overwrite the default value only `serviceType` present in the `values.yaml` file.

```suggestion
  {{- if .Values.clusterAgent.serviceType }}
  type: {{ .Values.clusterAgent.serviceType }}
  {{- end }}
```